src/helpers.rs

```diff
@@ -1312,3 +1312,15 @@ pub(crate) fn windows_check_buffer_size((success, len): (bool, u64)) -> u32 {
         u32::try_from(len).unwrap()
     }
 }
+
+pub(crate) fn get_arg<'tcx>(
+    ecx: &mut MiriInterpCx<'tcx>,
+    op: &OpTy<'tcx>,
+    ty: Ty<'tcx>,
+) -> InterpResult<'tcx, OpTy<'tcx>> {
+    let ty_layout = ecx.layout_of(ty)?;
+    if !op.layout.backend_repr.eq_up_to_validity(&ty_layout.backend_repr) {
+        throw_ub_format!("Invalid argument type: ABI mismatch");
+    }
```

Comment:
Hm, this is not quite the same logic as what we use to test ABI compatibility. That check cannot really be done on a per-argument / per-type basis; it works on `ArgAbi` which is part of `FnAbi`.

So, while this gives some extra coverage, it doesn't really go far enough, and we'd have to change everything completely again if we wanted to do the "proper" thing. I am not convinced it is worth investing a lot of effort into a half-way solution.